src/call/recording.js

```diff
@@ -0,0 +1,212 @@
+/*
+ * This file is responsible for all recording operations.
+ * This includes starting/stopping the recording,
+ * managing "Record" button state,
+ * and updating VCS text and image overlays for chat and emojis.
+ */
+
+import rand from './util.js';
+
+const visibleMessages = [];
+let reactionTimeout;
+
+/**
+ * Reflects recording state, such as whether
+ * the local participant is the owner of an ongoing recording
+ */
+export const recordingState = {
+  isRecordingOwner: false,
+};
+
+/**
+ * Sets up the Record button to toggle recording on/off.
+ * @param {DailyCall} callObject
+ */
+export function setupRecordToggle(callObject) {
+  const btn = getRecordBtn();
+  btn.onclick = () => {
+    btn.disabled = true;
+
+    const state = callObject.meetingSessionState();
+    const isRecording = state?.data?.isRecording;
```

Comment:
Minor note on the button state too -- since there's a bit of a delay, "Please wait..." text or a loading spinner might be better than just disabling the button because I wasn't sure what was happening at first. Depends how much time you want to put into details so consider it a nice-to-have!

Reply:
Good catch! I removed what I _thought_ was an unneeded state update at the last minute yesterday -.- But today I re-added it in a more appropriate/accurate place. Will add the spinner or text now as well.